builtin/providers/google/resource_compute_instance_template.go

```diff
@@ -403,16 +404,24 @@ func resourceComputeInstanceTemplateCreate(d *schema.ResourceData, meta interfac
 
 		if vp, okp := _scheduling["automatic_restart"]; okp {
 			instanceProperties.Scheduling.AutomaticRestart = vp.(bool)
+			forceSendFieldsScheduling = append(forceSendFieldsScheduling, "AutomaticRestart")
 		}
 
 		if vp, okp := _scheduling["on_host_maintenance"]; okp {
 			instanceProperties.Scheduling.OnHostMaintenance = vp.(string)
+			forceSendFieldsScheduling = append(forceSendFieldsScheduling, "OnHostMaintenance")
 		}
 
 		if vp, okp := _scheduling["preemptible"]; okp {
 			instanceProperties.Scheduling.Preemptible = vp.(bool)
+			forceSendFieldsScheduling = append(forceSendFieldsScheduling, "Preemptible")
+			if vp == true {
+				instanceProperties.Scheduling.OnHostMaintenance = "TERMINATE"
+				forceSendFieldsScheduling = append(forceSendFieldsScheduling, "OnHostMaintenance")
```

Comment:
You allocate the capacity for this slice as `3`, but you potentially add `4` items. This can cause runtime panic. I would recommend only placing `"OnHostMaintenance"` into `ForceSendFields` once, to avoid sending it twice.